### PR TITLE
UI: foundation for the new docs section of the website

### PR DESCRIPTION
**Pull request by avikalpg**
1. Added SideBar and RenderMarkdown components.
2. Created DocsSideBar view to use Sidebar in the documentation section
3. Added all the pages for the User Guide section of the documentation.

## Summary by CodeRabbit

- **New Features**
  - Introduced `RenderMarkdown` component for rendering markdown text.
  - Added `SideBar` component for sidebar navigation with collapsible items.
  - Added `BitbucketInstallation` and `GithubInstallation` components for setup guides.
  - Introduced `QuickStartGuide` component for Vibinex setup.
  - Added `TroubleshootingPages` component with dynamic routing for troubleshooting guides.
  - Introduced guides for starting a VM in the cloud and troubleshooting Docker command issues.

- **Improvements**
  - Enhanced documentation navigation with `DocsSideBar`.
  - Updated `next.config.js` for better handling of GitHub README images.

### CODE REVIEW

File: pages/docs/bitbucket-installation.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+import DocsSideBar from '../../views/docs/DocsSideBar';
+import MainAppBar from '../../views/MainAppBar';
+import Footer from '../../components/Footer';
+import Markdown from 'react-markdown';
+import remarkGfm from "remark-gfm";
+
+const BitbucketInstallation: React.FC = () => {
+	const installationMD = `
+## Sign in to Bitbucket
```
Comment (tapishr) [inline pages/docs/bitbucket-installation.tsx:10]: separate into .md file and push it into git? low logaf

File: pages/docs/index.tsx [hunk 2]
```diff
@@ -57,137 +37,69 @@ const Docs = ({ bitbucket_auth_url, image_name }: { bitbucket_auth_url: string,
 	React.useEffect(() => {
 		const anonymousId = getAndSetAnonymousIdFromLocalStorage()
 		rudderEventMethods?.track(getAuthUserId(session), "docs page", { type: "page", eventStatusFlag: 1, name: getAuthUserName(session) }, anonymousId)
+	}, [rudderEventMethods, session]);
 
-		const handleGitHubAppClick = () => {
-			rudderEventMethods?.track(getAuthUserId(session), "Install github app", { type: "link", eventStatusFlag: 1, source: "docs", name: getAuthUserName(session) }, anonymousId)
-		};
+	const introductionMD = `## What is Vibinex?
```
Comment (tapishr) [inline pages/docs/index.tsx:42]: same comment as above

Comment (tapishr) [reply pages/docs/index.tsx:42]: for all similar markdown content

File: pages/docs/index.tsx [hunk 3]
```diff
@@ -57,137 +37,69 @@ const Docs = ({ bitbucket_auth_url, image_name }: { bitbucket_auth_url: string,
...
-						</ol>
-					</AccordionContent>
-				</AccordionItem>
-			</Accordion>
+				<div className='sm:w-2/3 mx-auto mt-8 px-2 py-2'>
+					<RenderMarkdown markdownText={introductionMD} />
+					<RenderMarkdown markdownText={featuresMD} />
+					<ProductDemo />
+					<RenderMarkdown markdownText={architectureMD} />
+					{/* <Image src={architecturalDiagram(theme)} alt="Vibinex Setup" width={800} height={800} className="mx-auto my-4 w-full max-w-screen-md" /> */}
```
Comment (tapishr) [inline pages/docs/index.tsx:95]:
```suggestion
```

File: pages/docs/troubleshooting/index.tsx [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+		<div className="flex flex-col min-h-screen">
+			<MainAppBar />
+			<div className="flex-grow flex flex-row">
+				<DocsSideBar className="w-64" />
+				<div className="flex-grow p-8">
+					<h1 className="text-2xl font-bold mb-4">Troubleshooting Pages</h1>
+					<ul className="list-disc pl-4">
+						{pages.map((page) => (
+							<li key={page} className="mb-2">
+								<Link href={`/docs/troubleshooting/${page.replace('.tsx', '')}`} className="text-blue-500 hover:text-blue-700">
```
Comment (tapishr) [inline pages/docs/troubleshooting/index.tsx:24]: Cool 😎

File: pages/docs/quickstart.tsx
Comment (tapishr) [inline pages/docs/quickstart.tsx:1]: Didn't completely understand what's going on here, @avikalpg - a screenshot might be helpful

**Comment by tapishr** [comment]
Some of sonarcloud's  comments can also be addressed

File: pages/docs/quickstart.tsx
Comment (avikalpg) [reply pages/docs/quickstart.tsx:1]: quickstart page is currently the same as the original docs page.

**Comment by avikalpg** [comment]
Thanks for the review. I will incorporate your comments and push again tomorrow. I will also deeply verify the contents of the AI generated files.

**Comment by tapishr** [comment]
Incorporated this code and branch here - https://github.com/Alokit-Innovations/vibinex-server/pull/391
Closing this

**Pull request closed without merging**